lib/rucio/daemons/transmogrifier/transmogrifier.py

```diff
@@ -157,23 +158,24 @@ def is_matching_subscription(subscription, did, metadata):
     return True
 
 
-def select_algorithm(algorithm, rule_ids, params):
+def select_algorithm(algorithm, rule_ids, params, vo):
     """
     Method used in case of chained subscriptions
 
     :param algorithm: Algorithm used for the chained rule. Now only associated_site
                       associated_site : Choose an associated endpoint according to the RSE attribute assoiciated_site
     :param rule_ids: List of parent rules
     :param params: List of rules parameters to be used by the algorithm
+    :param vo: The vo on which to look for RSEs
     """
     selected_rses = {}
     if algorithm == 'associated_site':
         for rule_id in rule_ids:
             rule = get_rule(rule_id)
             logging.debug('In select_algorithm, %s', str(rule))
             rse = rule['rse_expression']
-            if rse_exists(rse):
-                rse_id = get_rse_id(rse)
+            if rse_exists(rse, vo=vo):
```

Comment:
As an alternative to using `vo` as an argument, we could use the lines below to get the VO from the RSE expression.
```
vo_match = re.search('(?<=vo=)[a-zA-Z0-9]{1,3}(?=&|$)', rse) #returns just 'tst' for input of 'vo=tst&(rse_expr)'
vo = vo_match if vo_match else 'def'
```
I think this is preferable as it avoids forcibly 'localising' `select_algorithm`.

Reply:
By avoiding localising it do you mean to prevent us from needing vo as an argument?
I didn't consider that the rse_expression from the rule would be in vo-ified format, but yeah this seems to work (though I think you need to add a `.group()` to the second line, since otherwise `vo` would be a match object rather than a string)?